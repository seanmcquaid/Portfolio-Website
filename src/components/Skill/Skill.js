import React from "react";

const Skill = props => {
    return(
        <div>Skill</div>
    )
};

export default Skill;