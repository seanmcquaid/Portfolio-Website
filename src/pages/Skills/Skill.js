import React from 'react';
import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';
import { FaAccessibleIcon } from 'react-icons/fa';

const Skill = ({ IconComponent, name }) => (
  <IconContainer id={`${name}Icon`} data-testid={`${name}Icon`}>
    <IconComponent aria-label={`${name} Icon`} />
    <IconName>{name}</IconName>
  </IconContainer>
);

const IconName = styled.label`
  margin: 0.25rem;
  font-size: 1rem;
  visibility: hidden;
  @media (max-width: 975px) {
    visibility: visible;
  }
`;

const fadeInAnimation = keyframes`
    0% { 
        opacity: 0; 
    } 
    100% { 
        opacity: 1; 
    } 
`;

const IconContainer = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 2rem;
  width: 4rem;
  margin: 0.75rem;
  @media (min-width: 975px) {
    &:hover ${IconName} {
      visibility: visible;
      transition: 0.25s;
      animation: ${fadeInAnimation} ease 1s;
      animation-iteration-count: 1;
      animation-fill-mode: forwards;
    }
  }
`;

Skill.propTypes = {
  IconComponent: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
};

Skill.defaultProps = {
  IconComponent: FaAccessibleIcon,
  name: 'Placeholder Name',
};

export default Skill;
