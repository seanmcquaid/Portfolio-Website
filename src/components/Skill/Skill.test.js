import React from "react";
import { render } from "@testing-library/react";
import Skill from "./Skill";
import { FaAccessibleIcon } from "react-icons/fa";

describe("<Skill/>", () => {
  it("Matches snapshot", () => {
    const props = {
      IconComponent: FaAccessibleIcon,
      name: "Placeholder Name",
    };
    const skill = render(<Skill {...props} />);
    expect(skill).toMatchSnapshot();
  });
});
