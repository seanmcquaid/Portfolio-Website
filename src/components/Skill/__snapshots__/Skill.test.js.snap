// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skill/> Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxirZ esJtMW"
        data-testid="Placeholder NameIcon"
        id="Placeholder NameIcon"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M423.9 255.8L411 413.1c-3.3 40.7-63.9 35.1-60.6-4.9l10-122.5-41.1 2.3c10.1 20.7 15.8 43.9 15.8 68.5 0 41.2-16.1 78.7-42.3 106.5l-39.3-39.3c57.9-63.7 13.1-167.2-74-167.2-25.9 0-49.5 9.9-67.2 26L73 243.2c22-20.7 50.1-35.1 81.4-40.2l75.3-85.7-42.6-24.8-51.6 46c-30 26.8-70.6-18.5-40.5-45.4l68-60.7c9.8-8.8 24.1-10.2 35.5-3.6 0 0 139.3 80.9 139.5 81.1 16.2 10.1 20.7 36 6.1 52.6L285.7 229l106.1-5.9c18.5-1.1 33.6 14.4 32.1 32.7zm-64.9-154c28.1 0 50.9-22.8 50.9-50.9C409.9 22.8 387.1 0 359 0c-28.1 0-50.9 22.8-50.9 50.9 0 28.1 22.8 50.9 50.9 50.9zM179.6 456.5c-80.6 0-127.4-90.6-82.7-156.1l-39.7-39.7C36.4 287 24 320.3 24 356.4c0 130.7 150.7 201.4 251.4 122.5l-39.7-39.7c-16 10.9-35.3 17.3-56.1 17.3z"
          />
        </svg>
        <p
          class="sc-AxjAm eTaZex"
        >
          Placeholder Name
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxirZ esJtMW"
      data-testid="Placeholder NameIcon"
      id="Placeholder NameIcon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M423.9 255.8L411 413.1c-3.3 40.7-63.9 35.1-60.6-4.9l10-122.5-41.1 2.3c10.1 20.7 15.8 43.9 15.8 68.5 0 41.2-16.1 78.7-42.3 106.5l-39.3-39.3c57.9-63.7 13.1-167.2-74-167.2-25.9 0-49.5 9.9-67.2 26L73 243.2c22-20.7 50.1-35.1 81.4-40.2l75.3-85.7-42.6-24.8-51.6 46c-30 26.8-70.6-18.5-40.5-45.4l68-60.7c9.8-8.8 24.1-10.2 35.5-3.6 0 0 139.3 80.9 139.5 81.1 16.2 10.1 20.7 36 6.1 52.6L285.7 229l106.1-5.9c18.5-1.1 33.6 14.4 32.1 32.7zm-64.9-154c28.1 0 50.9-22.8 50.9-50.9C409.9 22.8 387.1 0 359 0c-28.1 0-50.9 22.8-50.9 50.9 0 28.1 22.8 50.9 50.9 50.9zM179.6 456.5c-80.6 0-127.4-90.6-82.7-156.1l-39.7-39.7C36.4 287 24 320.3 24 356.4c0 130.7 150.7 201.4 251.4 122.5l-39.7-39.7c-16 10.9-35.3 17.3-56.1 17.3z"
        />
      </svg>
      <p
        class="sc-AxjAm eTaZex"
      >
        Placeholder Name
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
